src/inmanta/data/__init__.py

```diff
@@ -4026,7 +4039,17 @@ async def purge_logs(cls) -> None:
         for env in environments:
             time_to_retain_logs = await env.get(RESOURCE_ACTION_LOGS_RETENTION)
             keep_logs_until = datetime.datetime.now().astimezone() - datetime.timedelta(days=time_to_retain_logs)
-            query = "DELETE FROM " + cls.table_name() + " WHERE environment=$1 AND started < $2"
+            query = f"""
+            WITH non_halted_envs AS (
+              SELECT id FROM public.environment WHERE NOT halted
+            )
+            DELETE FROM {cls.table_name()}
+            WHERE environment = $1
+              AND started < $2
+              AND environment IN (
+                SELECT id FROM non_halted_envs
+            );
+            """
```

Comment:
It would be better to filter out the halted environment in the query that lists all environments.

```
environments = await Environment.get_list(halted=False)
```